__tests__/app.test.js

```diff
@@ -266,7 +265,79 @@ describe("GET /api/articles/:article_id/comments", () => {
         .get("/api/articles/hello/comments")
         .expect(400)
         .then(({ body }) => {
-          expect(body.msg).toBe("Something wrong with input");
+          expect(body.msg).toBe("Something wrong with input or body");
+        });
+    });
+  });
+
+  describe("POST /api/articles/:article_id/comments", () => {
+    test("POST 200: posts the user's comment", () => {
+      const newComment = { username: "butter_bridge", comment: "test comment" };
+      return request(app)
+        .post("/api/articles/1/comments")
+        .send(newComment)
+        .expect(201)
+        .then(({ body }) => {
+          const { postedComment } = body;
+          expect(postedComment).toBe(newComment.comment);
+        });
+    });
+
+    test("POST 400: returns an error message stating that the user is missing input data when no username", () => {
+      const newCommentNoUser = { comment: "test comment" };
+
+      return request(app)
+        .post("/api/articles/1/comments")
+        .send(newCommentNoUser)
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.msg).toBe("Something wrong with input or body");
+        });
+    });
+
+    test("POST 400: returns an error message stating that the yser is missing input data when no comment", () => {
+      const newCommentNoComment = { username: "testUser" };
```

Comment:
Problem with this test is you've sent testUser, so cant be sure that the 400 isnt coming from that, not the missing data